File: DKS-API/DTOs/SDtrLoginHistory.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SDtrLoginHistory : PaginationParams
    {
        public string systemName { get; set; }
        public string factoryId { get; set; }
        public string loginTimeS { get; set; }
        public string loginTimeE { get; set; }

    }
}

File: DKS-SPA/src/app/core/_models/s_dtr-login-history.ts
import { Pagination } from "./pagination";
import { utilityConfig } from "../utility/utility-config";

export class SDtrLoginHistory extends Pagination {
  systemName:string;
  factoryId: string;
  loginTimeS : string;
  loginTimeE : string;
  /**
   *default set of searching parameters
   */
  constructor() {
    super();
    this.systemName = ""; 
    this.factoryId = "";
    this.loginTimeS ="";
    this.loginTimeE ="";
    this.isPaging = true; //開分頁
  }

  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}
